packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/method_channel_google_maps_flutter.dart

```diff
@@ -22,10 +22,10 @@ class MethodChannelGoogleMapsFlutter extends GoogleMapsFlutterPlatform {
   /// /// Initializes the platform interface with [id].
   ///
   /// This method is called when the plugin is first initialized.
-  Future<void> init(int id) async {
+  Future<void> init(int id) {
     this._id = id;
     _channel = MethodChannel('plugins.flutter.io/google_maps_$_id');
-    await _channel.invokeMethod<void>('map#waitForMap');
+    _channel.invokeMethod<void>('map#waitForMap');
```

Comment:
Haha, in this case, you *do* want this method to be async and await for this, because you're waiting for the _channel.invokeMethod to return as a marker of it being done, sorry I wasn't more explicit!

In general, this advice is only applied when you do something like:

```
Future<Something> getSomething() async {
  Something value = await _anotherAsyncFunction();
  return value;
```

You can simplify to:

```
Future<Something> getSomething() {
  return _anotherAsyncFunction();
```

Because users of `getSomething` will have to `await` for the value anyways (because it's a `Future<Something>`)